hyp3_gamma/insar/getDemFileGamma.py

```diff
@@ -1,13 +1,18 @@
+import shutil
 from tempfile import NamedTemporaryFile
 
 from hyp3lib.execute import execute
 
 from hyp3_gamma.dem import get_geometry_from_kml, prepare_dem_geotiff
+from hyp3_gamma.insar.water_mask import apply_water_mask
 
 
-def get_dem_file_gamma(dem_image: str, dem_par: str, safe_dir: str, pixel_size: int):
+def get_dem_file_gamma(dem_image: str, dem_par: str, safe_dir: str, pixel_size: int, water_masking: bool):
     geometry = get_geometry_from_kml(f'{safe_dir}/preview/map-overlay.kml')
     with NamedTemporaryFile() as dem_tif:
         prepare_dem_geotiff(dem_tif.name, geometry, pixel_size)
+        if water_masking:
+            apply_water_mask(dem_tif.name, "tmp_masked_file.tif", safe_dir)
+            shutil.move("tmp_masked_file.tif", dem_tif.name)
         execute(f'dem_import {dem_tif.name} {dem_image} {dem_par} - - $DIFF_HOME/scripts/egm2008-5.dem '
-                f'$DIFF_HOME/scripts/egm2008-5.dem_par - - - 1', uselogging=True)
+                f'$DIFF_HOME/scripts/egm2008-5.dem_par - - - 0', uselogging=True)
```

Comment:
There's debugging to do here.  Changing the `zflg` parameter to `dem_import` causes values of zero in the source DEM are treated as NoData, causing *all* water (and some valid) land are *always* masked, regardless of the value of the `water_masking` parameter.

We should pick an alternate NoData value for the DEM (potentially impacting both RTC and InSAR), then mask water pixels using that new NoData value when `water_masking` is requested.

Reply:
new code inserts the nodata=minimum_valule_datatype into the dem.tif, then use this dme.tif in the gamma process. In the dem_import, we use execute(f'dem_import {dem_tif.name} {dem_image} {dem_par} - - $DIFF_HOME/scripts/egm2008-5.dem '  f'$DIFF_HOME/scripts/egm2008-5.dem_par - - - - -', uselogging=True).